### PR TITLE
feat: add search2 with ES only Artwork / Top pill functionality

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [FX-4564] 

### Description

⚠️ This functionality is hidden behind a FF `AREnableESOnlySearch`

- [x] Adds Search2 screen
- [x] only ES pills artwork / Top
- [x] some simple tests 

#### Screenshots

|Before|After|
|---|---|
|simulator_screenshot_15ACC87D-C0E5-4E62-96CD-8E25C0468036|Simulator Screen Shot - iPhone 14 Pro - 2023-01-30 at 14 00 39|

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add search2 with ES only Artwork / Top pill functionality, behind FF - gkartalis

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[FX-4564]: https://artsyproduct.atlassian.net/browse/FX-4564?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Search/Search2.tsx [hunk 1]
```diff
@@ -1,7 +1,199 @@
...
+
+      return unsubscribe
+    }
+  }, [navigation, searchProviderValues.inputRef.current])
+
+  return (
+    <SearchContext.Provider value={searchProviderValues}>
+      <ArtsyKeyboardAvoidingView>
+        <Flex p={2} pb={0}>
+          <SearchInput
```
Comment (gkartalis) [inline src/app/Scenes/Search/Search2.tsx:94]: **question:** should I move the input to its own component?

File: src/app/Scenes/Search/useSearchDiscoveryContentEnabled.ts [hunk 2]
```diff
@@ -2,9 +2,9 @@ import { useFeatureFlag } from "app/store/GlobalStore"
...
 export const useSearchDiscoveryContentEnabled = () => {
-  if (Platform.OS === "ios") {
-    return useFeatureFlag("AREnableSearchDiscoveryContentIOS")
-  }
+  const isIOS = Platform.OS === "ios"
 
-  return useFeatureFlag("AREnableSearchDiscoveryContentAndroid")
+  return useFeatureFlag(
+    isIOS ? "AREnableSearchDiscoveryContentIOS" : "AREnableSearchDiscoveryContentAndroid"
+  )
```
Comment (gkartalis) [inline src/app/Scenes/Search/useSearchDiscoveryContentEnabled.ts:9]: changed this to stop breaking the rules of hooks (hook was called conditionally)

File: src/app/Scenes/Search/Search2.tsx (hunk 1 above)
Comment (dblandin) [reply src/app/Scenes/Search/Search2.tsx:94]: The use of the component seems fine here but might be worth extracting the inline function for `onChangeText`.

**Review by dblandin — APPROVED** [review]
Nice start! 🎉

File: src/app/Scenes/Search/Search2.tests.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+describe("Search", () => {
+  const { renderWithRelay } = setupTestWrapper({
+    Component: () => <SearchScreen2 />,
+  })
+
+  it("should render a text input with placeholder and no pills", async () => {
+    renderWithRelay()
+
+    await flushPromiseQueue()
```
Comment (anandaroop) [inline src/app/Scenes/Search/Search2.tests.tsx:14]: **question (non-blocking):** I notice [some advice](https://github.com/artsy/eigen/blob/d563a5cc275feac237f5b9e9379edc8c02eac33d/docs/testing.md?plain=1#L72-L81) against using this in tests — not sure if it is still warranted in these instances?

File: src/app/Scenes/Search/Search2.tests.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  it("Top pill should be selected by default", async () => {
+    renderWithRelay()
+
+    await flushPromiseQueue()
+
+    const searchInput = screen.getByPlaceholderText("Search artists, artworks, galleries, etc")
+
+    fireEvent.changeText(searchInput, "text")
+
+    expect(screen.getByA11yState({ selected: true })).toHaveTextContent("Top")
```
Comment (anandaroop) [inline src/app/Scenes/Search/Search2.tests.tsx:45]: 😎 TIL `getByA11yState`

**Review by anandaroop — APPROVED** [review]
👍🏽

File: src/app/Scenes/Search/Search2.tests.tsx (hunk 3 above)
Comment (gkartalis) [reply src/app/Scenes/Search/Search2.tests.tsx]: Good point! These docs need a rewrite I started during Friday and will update them to the latest guidelines.

the `await flushPromiseQueue()` was practically making sure that the placeholder coming from suspense goes away, went ahead and did it properly 😅.

Thanks for the reminder ❤️

File: src/app/Scenes/Search/Search2.tsx [hunk 5]
```diff
@@ -1,7 +1,201 @@
...
+        </Flex>
+
+        <Flex flex={1} collapsable={false}>
+          {shouldStartSearching(searchQuery) ? (
+            <>
+              <Box pt={2} pb={1}>
+                <SearchPills
+                  ref={searchPillsRef}
+                  loading={false}
+                  pills={pillsArray}
```
Comment (dimatretyak) [inline src/app/Scenes/Search/Search2.tsx:125]: **suggestion**: we can just pass `DEFAULT_PILLS` here

Comment (gkartalis) [reply src/app/Scenes/Search/Search2.tsx:125]: will do it on the next PR that introduces the ES Pills ❤️

**Pull request merged**